[Bug]: Resiliency and chaos  test failing with TypeError exception

### Description
Resiliency tests failed with following threading issue 

```
2026-06-20 22:43:17 - request = <SubRequest 'resiliency_test_lifecycle' for <Function test_app_scale_on_storage_component_failure[STORAGECLUSTER_COMPONENT_FAILURES-CEPHFS_POD_FAILURES]>>

    @pytest.fixture(autouse=True)
    def resiliency_test_lifecycle(request):
        """
        Common lifecycle for all resiliency tests in this directory.
    
        - At test start: archive any existing Ceph crashes so the test starts clean.
        - During the entire test: background Ceph crash monitor every CEPH_CRASH_POLL_INTERVAL s.
        - finalizer: fail if Ceph crashes were introduced during the test.
        """
        try:
            ceph_status = CephStatusTool()
            ceph_status.archive_ceph_crashes()
            log.info(
                "Resiliency test lifecycle: archived pre-existing Ceph crashes (baseline)"
            )
        except Exception as e:
            log.warning(
                "Resiliency test lifecycle: could not archive pre-existing Ceph crashes: %s",
                e,
            )
    
        def _resiliency_finalizer():
            config = ResiliencyConfig()
            if not config.stop_when_ceph_crashed:
                return
            try:
                raise_if_ceph_crashes_detected(
                    CephStatusTool(),
                    "resiliency test finalizer",
                    poll_interval=0,
                )
            except AssertionError:
                raise
            except Exception as e:
                log.warning(
                    "Resiliency test lifecycle: could not run Ceph crash finalizer: %s",
                    e,
                )
    
        request.addfinalizer(_resiliency_finalizer)
    
        config = ResiliencyConfig()
>       with ceph_crash_monitor(
            enabled=config.stop_when_ceph_crashed,
            context="resiliency test",
        ):

tests/cross_functional/resilience/conftest.py:62: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib64/python3.11/contextlib.py:144: in __exit__
    next(self.gen)
ocs_ci/resiliency/resiliency_tools.py:334: in ceph_crash_monitor
    monitor.join(timeout=120)
/usr/lib64/python3.11/threading.py:1123: in join
    self._wait_for_tstate_lock(timeout=max(timeout, 0))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'Event' object is not callable") raised in repr()] CephCrashMonitor object at 0x7fcbe67014d0>
block = True, timeout = 120

    def _wait_for_tstate_lock(self, block=True, timeout=-1):
        # Issue #18808: wait for the thread state to be gone.
        # At the end of the thread's life, after all knowledge of the thread
        # is removed from C data structures, C code releases our _tstate_lock.
        # This method passes its arguments to _tstate_lock.acquire().
        # If the lock is acquired, the C code is done, and self._stop() is
        # called.  That sets ._is_stopped to True, and ._tstate_lock to None.
        lock = self._tstate_lock
        if lock is None:
            # already determined that the C code is done
            assert self._is_stopped
            return
    
        try:
            if lock.acquire(block, timeout):
                lock.release()
>               self._stop()
E               TypeError: 'Event' object is not callable

/usr/lib64/python3.11/threading.py:1141: TypeError
```

### Steps to Reproduce
1.
2.
3.

### Actual Behavior

### Expected Behavior

### Impact (likelihood of reproduction, impact on the cluster and on other tests, etc.)

### Screenshots (if applicable)

### Environment
- Test Suite(s):
- Platform(s):
- Version(s):
- OS:

### Additional Context

--- a/ocs_ci/resiliency/resiliency_tools.py
+++ b/ocs_ci/resiliency/resiliency_tools.py
@@ -252,7 +252,8 @@ class CephCrashMonitor(threading.Thread):
         super().__init__(daemon=True, name="ceph-crash-monitor")
         self.interval = interval
         self.context = context
-        self._stop = threading.Event()
+        # Avoid ``_stop`` — it shadows Thread._stop and breaks join() on Python 3.11+.
+        self._stop_event = threading.Event()
         self.crash_error = None
 
     def run(self):
@@ -285,12 +286,12 @@ class CephCrashMonitor(threading.Thread):
 
         if not _check_once():
             return
-        while not self._stop.wait(self.interval):
+        while not self._stop_event.wait(self.interval):
             if not _check_once():
                 break
 
     def stop(self):
-        self._stop.set()
+        self._stop_event.set()
 
     def raise_if_crash_detected(self):
         """Re-raise any crash detected by the background thread."""

